github/pulls.go

```diff
@@ -253,12 +253,16 @@ type PullRequestMergeResult struct {
 
 // PullRequestOptions lets you define how a pull request will be merged.
 type PullRequestOptions struct {
-	Squash bool
```

Comment:
But we don't have to. We may decide we want to, and that's fine. But it is easy to write:
```
type PullRequestOptions struct {
    Squash bool
    MergeMethod string
}
```
And then inside https://github.com/google/go-github/pull/446/files#diff-aafac035cce1f6a65b02566c5345b805R276
```
if options.Squash {
    mm := "squash"
    pullRequestBody.MergeMethod = &mm
}
```
Just pointing it out. I'm actually ok dropping OUR legacy API and break compile for our users. But its also pretty easy to be a friendly library if that's the path we wish to take.